### PR TITLE
permissions: add generator for handling groups enabled feature flag

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1672

### CODE REVIEW

File: invenio_users_resources/services/generators.py [hunk 1]
```diff
@@ -123,3 +125,28 @@ def query_filter(self, **kwargs):
                     return q_all & else_query
 
         return q_not_managed & then_query
+
+
+class GroupsEnabled(Generator):
```
Comment (anikachurilova) [inline invenio_users_resources/services/generators.py:130]: Moved from https://github.com/inveniosoftware/invenio-communities/pull/1127/files#diff-2476adbb5df07ac52ab8b38741dacd9f144d1ee77dbcef244085581885d6c530L346

File: invenio_users_resources/config.py [hunk 2]
```diff
@@ -240,3 +240,6 @@ class OrgPropsSchema(Schema):
 
 USERS_RESOURCES_DOMAINS_ORG_SCHEMA = OrgPropsSchema
 """Domains organisation schema config."""
+
+USERS_RESOURCES_GROUPS_ENABLED = True
```
Comment (ntarocco) [inline invenio_users_resources/config.py:244]: This needs to be very well documented and our colleagues from Zenodo need to put it to `False`, let's don't forget it :)

File: invenio_users_resources/services/generators.py [hunk 3]
```diff
@@ -11,6 +11,8 @@
 """Permission generators for users and groups."""
 
 
+from flask import current_app
```
Comment (ntarocco) [inline invenio_users_resources/services/generators.py:14]: what is probably missing is to the change the groups service config, and to change the `can_read` to take into account this new flag.

Comment (anikachurilova) [reply invenio_users_resources/services/generators.py:14]: Done!

File: invenio_users_resources/services/generators.py [hunk 4]
```diff
@@ -123,3 +125,28 @@ def query_filter(self, **kwargs):
...
+    A system process is allowed to do anything.
+    """
+
+    def __init__(self, *need_groups_enabled_types):
+        """Types that need the groups enabled."""
+        self.need_groups_enabled_types = need_groups_enabled_types
+
+    def excludes(self, member_types=None, **kwargs):
+        """Preventing needs."""
+        member_types = member_types or {"group"}
```
Comment (anikachurilova) [inline invenio_users_resources/services/generators.py:145]: Note: added default member_types (different from the old code)

**Review by ntarocco — APPROVED** [review]

**Pull request merged**